apps/src/Sound.js

```diff
@@ -340,10 +371,17 @@ Sound.prototype.preloadViaWebAudio = function (filename, onPreloadedCallback) {
   request.responseType = 'arraybuffer';
   var self = this;
   request.onload = function () {
-    self.audioContext.decodeAudioData(request.response, function (buffer) {
-      onPreloadedCallback(buffer);
-      self.onSoundLoaded();
-    });
+    if (request.status === 200) {
+      self.audioContext.decodeAudioData(request.response, function (buffer) {
+        onPreloadedCallback(buffer);
+        self.onSoundLoaded();
+      });
+    } else {
+      self.handleLoadFailed(request.status);
+    }
+  };
+  request.onerror = function () {
+    self.handleLoadFailed(request.status);
```

Comment:
I don't remember this well. Do we still get a reasonable `request.status` code in the case on `.onerror`?

Reply:
My observations in Chrome are that we get status `0` here. In my testing this only happens when I disconnect the network. The `403` that we respond to in Dance.js is actually coming in through the `request.onload` path above.